Adds the Vancouver-class Salvage Freighter

Ratvarr opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Adds the Vancouver-class Salvage Freighter, and its internal vessel, the Esquimalt-class.
"When faced with a lot of salvage, and not a lot of time, most vessels scrap the key parts, and leave a crab-shell of hull and wires ripe for the taking. So, with time and supplies becoming an issue, the ever resolute and probably (not) reliable Miskilamo Spacefaring produced the Vancouver-class, and its internal Esquimalt-class shuttle, so that you can stay in the field for longer, and get into those tight spaces."
CREW: Captain
Foreman(2ic)
Vessel Engineer (dedicated stay at home engineer)
3 Salvager
Security Detail 
2 Deckhand
<details><summary>GEAR</summary>
<p>
APLU, salvage saw for it, and a drill+clamp
EHF Beacon
Emitter+Rapid Cable Layer
Anti-Radiation Foam
Random scrap from the last haul
A subshuttle
Conflag, 2 boxes of ammo for it
2 E11, 2 E10, 2 standard cells spare overall.
Etherbor SG8+spare etherbor cell (standard)
basic armour, and a machete for the sec detail, and a stimulant medipen (if  you dare)
2 angle grinders
</p>
</details> 

comes with an amazing, top of the line smokers lounge to comply with some regulation somewhere, some concerning blood in the cargo bay, and a lot of potatoes ready to be ~~fermeneted into~~ baked.
somewhat claustrophobic in sections but has enough space for any random projects
subshuttle works, vessel works, have tested to make sure it doesn't brutally explode into a thousand billion pieces unless you fly into asteroids

<details><summary>IMAGES</summary>
<p>
VANCOUVER-CLASS:

![image](https://github.com/user-attachments/assets/4d2e0fc7-74b7-46e6-b053-79c624872c57)

![image](https://github.com/user-attachments/assets/37c96312-8eac-4bdf-a451-8acabf605335)

![image](https://github.com/user-attachments/assets/3558cd0a-ed32-4c5a-b5d8-925a8e6b9346)

![image](https://github.com/user-attachments/assets/e8691af2-2bae-44d0-ab2b-35afc5fa8509)
s

ESQUIMALT-CLASS:

![image](https://github.com/user-attachments/assets/c793d293-3fa3-4d6c-b2d2-b209664e99aa)

![image](https://github.com/user-attachments/assets/0a86a485-ca5f-48ff-99e6-ee9fd3e2ed9f)

![image](https://github.com/user-attachments/assets/3c152c8f-dfd5-4fe3-9676-ad5348093f65)


</p>
</details> 

## Why It's Good For The Game
subshuttle indie ship good I think. 
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

:cl:
add: Vancouver-class Salvage Freighter
add: Esquimalt-class subshuttle
/:cl:

Ratvarr committed:
Create independent_vancouver.dmm

Ratvarr committed:
dont want to lose my progress should my PC explode into a billion shards

Ratvarr committed:
Merge branch 'shiptest-ss13:master' into sir,-another-indie-ship-has-hit-the-dev-forums

Ratvarr committed:
okay

Ratvarr committed:
okay

Ratvarr committed:
makes it work

Ratvarr committed:
ready for review

Ratvarr committed:
actually ready

Ratvarr committed:
Update independent_esquimalt.dmm

Ratvarr committed:
Merge branch 'master' into sir,-another-indie-ship-has-hit-the-dev-forums

Ratvarr committed:
should fix this

Ratvarr committed:
Merge branch 'sir,-another-indie-ship-has-hit-the-dev-forums' of https://github.com/Ratvarr/Shiptest-Torment-Nexus into sir,-another-indie-ship-has-hit-the-dev-forums

Ratvarr committed:
Update independent_vancouver.dmm

Ratvarr commented:
okay dokey i know the ship_placement_test *is* failing Forever but thats just because of how the SMES terminals are setup which has the PACMAN units placed directly onto their terminals, thus making it go "we have no wires !!!!!" but they do power

Ratvarr committed:
okay should fix checks and also work with the new om5 thing

Ratvarr committed:
Update independent_vancouver.dmm

Ratvarr committed:
Update independent_vancouver.dmm

Ratvarr commented:
okay well i fixed it all we win

Gristlebee reviewed (changes requested):
- The blast doors are two different types on your subshuttle
![image](https://github.com/user-attachments/assets/0d67bc35-7ace-4c92-9eea-0cc985d04c5a)
- Your subshuttle is wired incorrectly. The power directly attaches to the SMES output wires (red) rather than it's input wires (yellow)
![image](https://github.com/user-attachments/assets/fa542722-37e4-43e2-b54f-f92c781ed498)
- The subshuttle hangar is very densely packed and hard to navigate. I'd get rid of some of the salvagables on the right side, so there's a tile worth of clearance between them. 
- I think it would also help a lot with the flow of the bay if the either/both the top and bottom paths were two tile wide. A suggestion I have is have the subshuttle exit open from the bottom side, and have the bottom part of the hangar have two tile clearance. If this is meant to be a salvage ship, having elbow room to move your salvage out of the ship is going to make things feel a lot better. 
![image](https://github.com/user-attachments/assets/1d6ebf80-a2f7-48f4-8c63-c8c21bc561ef)
- The foreman's belt could be the subtype loaded with all the proper gear instead of the empty one without offsetting the ships balance so all the tools don't start dumped out.  If you want to keep the foreman toolbelt empty, have the locker start with a basic toolbox instead so everything isn't loose and can be moved to the belt as needed.
- Armory is very dark
![image](https://github.com/user-attachments/assets/91ad50c7-ae36-4c2d-a5ff-149b42772ba5)
- The armory frankly is very claustrophobic. If that's the intended vibe, you achieved it. Otherwise, a potential suggestion is to have a windows in the circled portion to overlook the hangar bay, and have it be something like an office/prep space a la the shetland or the miner's room on the riggs.
![image](https://github.com/user-attachments/assets/b5dc3b55-297f-4d14-8aad-a7a922ae5592)
- Dorms is quite dark on the left side.
![image](https://github.com/user-attachments/assets/84371f55-f7e4-4bce-9907-bd58d7804720)
- I'd have these start in a wall locker if you're going to have them scattered like this. It's not messy in an aesthetically pleasing way.
![image](https://github.com/user-attachments/assets/d7b5cfa0-c23f-4703-a95a-fa804b1f58b3)
- This feels rather dark for a main thoroughfare.
![image](https://github.com/user-attachments/assets/8fabddff-12ab-45c4-9e5e-68c8732f6aae)
- The floor tile transition between the plated catwalk and techfloor looks indistinct. Play around with decals or different tile types to make them stand out more against each other.
![image](https://github.com/user-attachments/assets/7a4b725a-c01a-465b-889e-56589a88cd33)
- The captain's quarters has no lights.
![image](https://github.com/user-attachments/assets/9e4d8e0c-bd3d-4644-b604-cfc6f4468f6d)
- The shower feels dark. Add a second light inside the shower stall.
![image](https://github.com/user-attachments/assets/5abf7253-c03d-404b-8085-692cd2f2c760)

Ratvarr commented:
blast doors are specifically chosen to be two different types as a stylistic choice
shuttle moved down, and re-wired correctly
foreman belt kept empty, removed other tools and added mech+elec toolbox to locker
dorms specifically setup to be darker on the left, to have you turn on the the lamp
![image](https://github.com/user-attachments/assets/158d7a19-41c2-4476-a692-8ee126206f77)

main throufare lit up better, only minor darkness outside the bathroom in a minor degree
bathroom second light installed
replaced engine room tiles with techfloormaint
![image](https://github.com/user-attachments/assets/89450759-54f6-4d4d-bfad-9bf6df659a9a)

captains quarters kept without light on wall, with the desk lamp being the source of light (namely because the room is apart of the crew eating/rec room, and its kind of extremely small to have a proper APC+etc
armoury installed with windows+shutters to overlook the hanger, but its also not meant to be the gangster office its closer to a broom closet with your 12 gauge to keep the Tax Man off your ship instead of the indie merc larp room which makes you look more like a office worker than a security detail
ignore how awful my localhost character is I forgot to change off her

Ratvarr committed:
forgot to commit award

Gristlebee reviewed (changes requested):
- Missed this on the first pass, but the grinder crate should have some kind of eye  protection.
- The foreman's locker should have spare clothing as well.
- The subshuttle is still wired incorrectly, as it doesn't line up with the main grid now.
![image](https://github.com/user-attachments/assets/5273217a-0867-44e1-8889-cda43bc4b1f3)
- I do not feel it the different blast doors work as an aesthetic choice. I feel they clash rather than work together as an intentional choice to my eyes. If you can get another maptainer to say yes to it, I'll accept it, but it's a no from me for now. I'd rather you have the blastdoors/shutters match.
![image](https://github.com/user-attachments/assets/4fdf434e-e1ec-4306-956a-83d5a8cc4c0d)

Ratvarr committed:
im so good

Ical92 commented:
![image](https://github.com/user-attachments/assets/fcd37b77-f151-4545-80fd-ef161540676f)


I personally think this gap here makes the shuttle hangar feel a bit too structurally detached from the rest of the ship. Could just fill the gap in, or maybe put some supporting lattices around it?

Ratvarr committed:
good suggestion

Ratvarr committed:
Merge branch 'master' into sir,-another-indie-ship-has-hit-the-dev-forums

Ratvarr committed:
yeahhhaha

Ratvarr committed:
bratty cargo bay doors get

Ratvarr committed:
bad fix but whateva

Crystal1Blitz commented:
Hey, Communism here, great ship, but I would like to say that having a wire eating mouse with miskilamo wiring and the plated catwalks is kinda bad, he walked out, ate a wire, and I thought I fixed it until it turned out the wires led to nothing.

meemofcourse commented:
> Hey, Communism here, great ship, but I would like to say that having a wire eating mouse with miskilamo wiring and the plated catwalks is kinda bad, he walked out, ate a wire, and I thought I fixed it until it turned out the wires led to nothing.

can we put this on starboard

Maksoplo commented:
> > Hey, Communism here, great ship, but I would like to say that having a wire eating mouse with miskilamo wiring and the plated catwalks is kinda bad, he walked out, ate a wire, and I thought I fixed it until it turned out the wires led to nothing.
> 
> can we put this on starboard

🚢

Ratvarr commented:
> Hey, Communism here, great ship, but I would like to say that having a wire eating mouse with miskilamo wiring and the plated catwalks is kinda bad, he walked out, ate a wire, and I thought I fixed it until it turned out the wires led to nothing.

i have to make slight fun of you over the fact that you let the mouse walk out of the captains room, into the crew-storage-fore-whatever-spot, and it chewed through the SINGLE redundant wire in the entire ship, or it ate the heart in the cargo bay

Ratvarr committed:
bye bye mousy

Ratvarr committed:
Update independent_vancouver.dmm

Ratvarr committed:
Merge branch 'master' into sir,-another-indie-ship-has-hit-the-dev-forums

Signed-off-by: Ratvar <61395690+Ratvarr@users.noreply.github.com>

generalthrax commented:
indentation on the esquimalt is preventing testmerge
![image](https://github.com/user-attachments/assets/00e3bc13-726d-420c-9d41-193d2126a810)

ritorizo commented on code/datums/shuttles.dm:
```diff
@@ -459,9 +459,16 @@
+/datum/map_template/shuttle/subshuttles/esquimalt
+	file_name = "independent_esquimalt"
+	name = "Esquimalt-class Salvage Pod"
+  token_icon_state = "ship_tiny_generic"
```
```suggestion
	token_icon_state = "ship_tiny_generic"
```

Ratvarr committed:
Update code/datums/shuttles.dm

Co-authored-by: ritorizo <41831966+ritorizo@users.noreply.github.com>
Signed-off-by: Ratvar <61395690+Ratvarr@users.noreply.github.com>

thearbiber commented:
Apologies, but we will not be accepting this PR currently. There are some fundamental issues with the current interior layout, along with lowering interests of more miski ships in the maptainer team. Independent scrappers also are a full niche at this point. Thank you for your contribution, still.
